product_sequence/models/product_category.py

```diff
@@ -38,22 +38,31 @@ def _prepare_ir_sequence(self, prefix):
         }
         return vals
 
+    def _get_or_create_sequence(self, prefix):
+        seq_vals = self._prepare_ir_sequence(prefix)
+        sequence = self.env["ir.sequence"].search(
+            [("code", "=", seq_vals.get("code")), ("prefix", "=", prefix)]
+        )
+        if not sequence:
+            sequence = self.env["ir.sequence"].create(seq_vals)
+        return sequence
+
     def write(self, vals):
-        prefix = vals.get("code_prefix", False)
-        if prefix:
-            for rec in self:
-                if rec.sequence_id:
-                    rec.sudo().sequence_id.prefix = prefix
-                else:
-                    seq_vals = self._prepare_ir_sequence(prefix)
-                    vals["sequence_id"] = self.env["ir.sequence"].create(seq_vals)
+        if "code_prefix" in vals:
+            prefix = vals.get("code_prefix")
+            if prefix:
```

Comment:
This change makes the diff larger but does not seems to be needed.

Reply:
We actually need to check the existence of `code_prefix` key in the values dictionary that is provided, to remove the link to the sequence if the prefix is empty. (or at least I don't see any other way for a normal user that doesn't have rights on `ir.sequence` to remove this link)